File: java/codechef/MMAX.java
import java.util.*;
import java.lang.Math;

class MMAX {
    static Scanner sc = new Scanner(System.in);

    public static void main(String[] args) {
        int t = sc.nextInt();
        for(int i=0; i<t; i++) {
            long n = sc.nextLong();
            long k = sc.nextLong();
            long num1 = 0;
            long num2 = 0;
            if (n>2) {
                if (n>=k) {
                    num1 = k;
                    num2 = n-num1;
                }
                else {
                    num1 = k%n;
                    num2 = n-num1;
                }
                
                long sum2 = 2 * Math.min(num1, num2);
                System.out.println(sum2);
            }
            else {
                System.out.println(k%2);
            }
        }
    }
}

File: python/codechef/MMAX.py
t = int(input())
for i in range(t):
    n = int(input())
    k = int(input())
    num1 = 0
    num2 = 0
    if (n>2):
        if (n>=k):
            num1 = k
            num2 = n-num1

        else:
            num1 = k%n
            num2 = n-num1
        if num1==num2 :
            sum2 = 2*num1 - 1
        else :
            sum2 = 2 * min(num1, num2)
        print(sum2)
    
    else:
        print(k%2)
